### PR TITLE
Vcluster/1857 terms

### PR DESCRIPTION
**Pull request by ckr123**

### CODE REVIEW

File: domains/auth/Auth.API/API/Repositories/UserRepository.cs [hunk 1]
```diff
@@ -31,4 +39,12 @@ public async Task RemoveUserAsync(User user)
     }
 
     public async Task<IEnumerable<User>> GetUsersByTinAsync(string tin) => await dataContext.Users.Where(x => x.Company != null && x.Company.Tin == tin).Include(u => u.UserRoles).ToListAsync();
+    public async Task<User> UpdateTermsAccepted(User user, DecodableUserDescriptor descriptor)
+    {
+        dataContext.Users.Update(user);
+        await publishEndpoint.Publish(new OrgAcceptedTerms((Guid)user.Id!, descriptor.Organization?.Tin, descriptor.Id));
+
+        await dataContext.SaveChangesAsync();
+        return user;
```
Comment (tnickelsen) [inline domains/auth/Auth.API/API/Repositories/UserRepository.cs:48]: Not pretty to have the repo publish events. Can you expose SaveChangesAsync isntead? 
On a sidenote we need UnitOfWork ideally for this

File: domains/measurements/Measurements.API/API/Program.cs [hunk 2]
```diff
@@ -16,7 +16,7 @@
 try
 {
     Log.Information("Starting server.");
-    WebApplication app = configuration.BuildApp();
+    WebApplication app = configuration.BuildApp(args);
```
Comment (tnickelsen) [inline domains/measurements/Measurements.API/API/Program.cs:19]: Why is this needed?

File: domains/measurements/Measurements.API/Tests/Fixtures/TestServerFixture.cs [hunk 3]
```diff
@@ -31,8 +37,13 @@
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Hosting;
 using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Testcontainers.PostgreSql;
+using Testcontainers.RabbitMq;
 using Tests.Fixtures.TestServerHelpers;
+using Xunit;
 using Xunit.Abstractions;
+using RabbitMqContainer = Testcontainers.RabbitMq.RabbitMqContainer;
```
Comment (tnickelsen) [inline domains/measurements/Measurements.API/Tests/Fixtures/TestServerFixture.cs:46]: Many usings added but they are not used?

File: domains/measurements/Measurements.API/API/MeteringPoints/Api/Consumer/TermsConsumer.cs [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+using API.Options;
+using EnergyOrigin.IntegrationEvents.Events.Terms.V1;
+using MassTransit;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Options;
+using Relation.V1;
+
+namespace API.MeteringPoints.Api.Consumer;
+
+public class TermsConsumer(
```
Comment (tnickelsen) [inline domains/measurements/Measurements.API/API/MeteringPoints/Api/Consumer/TermsConsumer.cs:14]: Have you tested in a vcluster?

File: domains/measurements/Measurements.API/Tests/MeteringPoints/Api/MeteringPointControllerTests.cs [hunk 5]
```diff
@@ -62,34 +63,42 @@ public async Task GetMeteringPoints()
...
+        var clientMock = _factory.Services.GetRequiredService<Meteringpoint.V1.Meteringpoint.MeteringpointClient>();
         clientMock.GetOwnedMeteringPointsAsync(Arg.Any<Meteringpoint.V1.OwnedMeteringPointsRequest>())
             .Returns(mockedResponse);
-        _serverFixture.ConfigureTestServices += services =>
-        {
-            var mpClient = services.Single(d =>
-                d.ServiceType == typeof(Meteringpoint.V1.Meteringpoint.MeteringpointClient));
-            services.Remove(mpClient);
-            services.AddSingleton(clientMock);
-        };
```
Comment (tnickelsen) [inline domains/measurements/Measurements.API/Tests/MeteringPoints/Api/MeteringPointControllerTests.cs:74]: Why was this needed?

File: domains/measurements/Measurements.API/Tests/MeteringPoints/Api/MeteringPointControllerTests.cs [hunk 6]
```diff
@@ -4,34 +4,37 @@
...
-using Tests.Fixtures;
+using Tests.TestContainers;
 using VerifyTests;
 using VerifyXunit;
 using Xunit;
 
 namespace Tests.MeteringPoints.Api;
 
-public class MeteringPointControllerTests : MeasurementsTestBase, IDisposable
+public class MeteringPointControllerTests : IClassFixture<CustomMeterPointWebApplicationFactory<Startup>>, IClassFixture<PostgresContainer>
```
Comment (tnickelsen) [inline domains/measurements/Measurements.API/Tests/MeteringPoints/Api/MeteringPointControllerTests.cs:22]: Why was it needed to make a custom?

**Review by tnickelsen — APPROVED** [review]

**Review by tnickelsen — APPROVED** [review]

File: domains/auth/Auth.API/API/Program.cs [hunk 7]
```diff
@@ -118,7 +115,21 @@
...
+        var options = sp.GetRequiredService<IOptions<RabbitMqOptions>>().Value;
+        var factory = new ConnectionFactory
+        {
+            HostName = options.Host,
+            Port = options.Port ?? 0,
+            UserName = options.Username,
+            Password = options.Password,
+            AutomaticRecoveryEnabled = true,
+        };
+        o.Connection = factory.CreateConnection();
```
Comment (TopSwagCode) [inline domains/auth/Auth.API/API/Program.cs:131]: Note to self - Lookup HealthCheck
